frontend/views/containers/payments/PaymentsMixin.js

```diff
@@ -19,55 +19,30 @@ const PaymentsMixin: Object = {
         todo: cloneDeep(this.ourPayments?.todo || [])
       }
 
-      const paymentsByPeriodKey = `paymentsByPeriod/${this.ourUsername}/${this.currentGroupId}`
-      const paymentsByPeriod = await sbp('gi.db/archive/load', paymentsByPeriodKey) || {}
-      const paymentsKey = `payments/${this.ourUsername}/${this.currentGroupId}`
-      const payments = await sbp('gi.db/archive/load', paymentsKey) || {}
+      const historicalSentOrReceivedPaymentsKey = `sentOrReceivedPayments/${this.ourUsername}/${this.currentGroupId}`
+      const historicalSentOrReceivedPayments = await sbp('gi.db/archive/load', historicalSentOrReceivedPaymentsKey) ||
+        { sent: [], received: [] }
 
-      for (const period of Object.keys(paymentsByPeriod).sort().reverse()) {
-        const { paymentsFrom } = paymentsByPeriod[period]
-        for (const fromUser of Object.keys(paymentsFrom)) {
-          for (const toUser of Object.keys(paymentsFrom[fromUser])) {
-            if (toUser === this.ourUsername || fromUser === this.ourUsername) {
-              const receivedOrSent = toUser === this.ourUsername ? 'received' : 'sent'
-              for (const hash of paymentsFrom[fromUser][toUser]) {
-                const { data, meta } = payments[hash]
-                paymentsInTypes[receivedOrSent].push({ hash, data, meta, amount: data.amount, username: toUser })
-              }
-            }
-          }
-        }
+      return {
+        sent: [...recentSentOrReceivedPayments.sent, ...historicalSentOrReceivedPayments.sent],
+        received: [...recentSentOrReceivedPayments.received, ...historicalSentOrReceivedPayments.received],
+        todo: recentSentOrReceivedPayments.todo
       }
-      const sortPayments = payments => payments
-        .sort((f, l) => f.meta.createdDate > l.meta.createdDate ? 1 : -1)
-      paymentsInTypes.sent = sortPayments(paymentsInTypes.sent)
-      paymentsInTypes.received = sortPayments(paymentsInTypes.received)
-
-      return paymentsInTypes
     },
     async getPaymentDetailsByPeriod (period: string) {
-      let detailedPayments = {}
       if (Object.keys(this.groupPeriodPayments).includes(period)) {
         const paymentHashes = this.paymentHashesForPeriod(period) || []
-        detailedPayments = paymentHashes.map(hash => this.currentGroupState.payments[hash])
-      } else {
-        const paymentsByPeriodKey = `paymentsByPeriod/${this.ourUsername}/${this.currentGroupId}`
-        const paymentsByPeriod = await sbp('gi.db/archive/load', paymentsByPeriodKey) || {}
-        const paymentHashes = paymentHashesFromPaymentPeriod(paymentsByPeriod[period])
-
-        const paymentsKey = `payments/${this.ourUsername}/${this.currentGroupId}`
-        const payments = await sbp('gi.db/archive/load', paymentsKey) || {}
-        for (const hash of paymentHashes) {
-          detailedPayments[hash] = payments[hash]
-        }
+        return paymentHashes.map(hash => this.currentGroupState.payments[hash])
```

Comment:
@Silver-IT There is two return statements in this function but apparently their types are different. it returns an array here whereas the one in below L:45 returns an object. Did you probably want to transform the array here to an object equivalent like `Object.fromEntries(paymentHashes.map(hash => [hash, this.currentGroupState.payments[hash]]))`

Reply:
It's really good catch. Thanks @SebinSong.